PROJECT-CLEARWATER-MIB

```diff
@@ -2447,4 +2450,237 @@ mementoGroup OBJECT-GROUP
+
+astaireConnectionTable OBJECT-TYPE
+    SYNTAX SEQUENCE OF AstaireConnectionTableEntry
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Resynchronization statistics on a per-connection basis."
+    ::= { astaire 6 }
+
+astaireConnectionTableEntry OBJECT-TYPE
+    SYNTAX      AstaireConnectionTableEntry
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Resynchronization statistics for a specific connection."
+    INDEX { astaireConnectionInetAddrType,
+            astaireConnectionInetAddr      }
+    ::= { astaireConnectionTable 1 }
+
+AstaireConnectionTableEntry ::= SEQUENCE
+{
+  astaireConnectionInetAddrType           InetAddressType,
+  astaireConnectionInetAddr               InetAddress,
+  astaireConnectionBucketsNeedingResync   Unsigned32,
+  astaireConnectionBucketsResynchronized  Unsigned32,
+  astaireConnectionBucketTable            SEQUENCE OF astaireConnectionBucketTableEntry
+}
+
+astaireConnectionInetAddrType OBJECT-TYPE
+    SYNTAX      InetAddressType
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Inet address type of the connection peer"
+    ::= { astaireConnectionTableEntry 1 }
+
+astaireConnectionInetAddr OBJECT-TYPE
+    SYNTAX      InetAddress (SIZE (1..16))
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Inet address of the connection peer"
+    ::= { astaireConnectionTableEntry 2 }
+
+astaireConnectionBucketsNeedingResync OBJECT-TYPE
+    SYNTAX      Unsigned32
+    MAX-ACCESS  read-only
+    STATUS      current
+    DESCRIPTION "Count of buckets needing to be synchronized over this
+                 connection for the current resynchronization operation."
+    ::= { astaireConnectionTableEntry 3 }
+
+astaireConnectionBucketsResynchronized OBJECT-TYPE
+    SYNTAX      Unsigned32
+    MAX-ACCESS  read-only
+    STATUS      current
+    DESCRIPTION "Number of buckets already resynchronized over this connection
+                 during the current resynchronization operation."
+    ::= { astaireConnectionTableEntry 4 }
+
+astaireConnectionBucketTable OBJECT-TYPE
+    SYNTAX SEQUENCE OF AstaireConnectionBucketTableEntry
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Resynchronization statistics on a per-connection and
+                 per-bucket basis."
+    ::= { astaireConnectionTableEntry 5 }
+
+astaireConnectionBucketTableEntry OBJECT-TYPE
+    SYNTAX      AstaireConnectionBucketTableEntry
+    MAX-ACCESS  not-accessible
+    STATUS      current
+    DESCRIPTION "Resynchronization statistics for a specific connection and
+                 bucket."
+    INDEX { astaireConnectionBucketId }
+    ::= { astaireConnectionBucketTable 1 }
+
+AstaireConnectionBucketTableEntry ::= SEQUENCE
+{
+  astaireConnectionBucketId                     Unsigned32,
+  astaireConnectionBucketEntriesResynchronized  Unsigned32,
+  astaireConnectionBucketDataResynchronized     Unsigned32
+}

```

Comment:
How long do these stats last?  The connection disappears after any related resynch operations complete (and might be recreated to handle reads from backups).